node/tessel-export.js

```diff
@@ -1006,11 +1017,15 @@ Tessel.SPI = function(params, port) {
 };
 
 Tessel.SPI.prototype.send = function(data, callback) {
-  this._port.cork();
+
   this.chipSelect.low();
+  this._port.delay(this.chipSelectDelayUs);
+  this._port.cork();
```

Comment:
Take a look at this gist: https://gist.github.com/rwaldron/4609336bb152388260c753d8e176fa8a

The average execution of _port.cork(), over 10 measurements is about 2.74ms, which means that an SPI object created with a chipSelectDelayUs that has any value less than 2740μs is always unnecessary (and indeed detrimental), because the call to cork has already taken that amount of time. I add that it's detrimental, because it's adding 2.5-3ms to the execution—even when the delay is zero.